module-4/backend/week_four.md

```diff
@@ -5,15 +5,36 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What's your favorite project management tool?
+  * Waffle. I like Pivotal Tracker too but I find Waffle to be more flexible which generally suits my needs better, and I find it easier to sync to GitHub.
+  
 2. Why do we create staging environments?
+  * Staging is meant to exactly replicate the production environment (OS, database, etc.) so that we can ensure our code will work in production before we actually deploy it.
+  
 3. What are the characteristics of a good README (in your opinion)?
+  * A good README contains everything a user would need to know to get the code running on their machine. The best READMEs may include screenshots of what to expect the app to look like, as well as links to give the reader more information when necessary.
+  
 4. What's one main improvement you're going to make to your code regarding accessibility issues?
+  * Using semantic HTML tags when possible, as well as using the tabindex tag, will both help users using screenreaders. Using sans-seriff fonts with clear distinctions between font colors and background colors will also help visually impaired users.
+  
 5. What are some basic security concerns to be aware of when building applications?
+  * Cross-site scripting, mass assignment, and SQL injection are some of the top concerns.
+  
 6. Why is continuous integration helpful/important?
+  * CI allows for code to be merged frequently, which enables bugs or issues to be found and addressed more quickly.
+  
 7. What are some continuous integration tools?
+  * HoundCI and TravisCI
 
 #### Review  
 
 8. What are some characteristics of "good" git workflow?
+  * Logical branch names with commits that encompass logical units of work, descriptive commits and PRs, and never merging your own PR. When merging someone's PR, address any concerns or questions via comments on code lines as well as comments on the PRs themselves.
```

Comment:
good response - i'd love to hear about why we want logical branch names and commits that encompass logical units of work? what purpose does this serve? why do we even care about following "good" git (or version) workflow?